go/enclave/main/main.go

```diff
@@ -4,16 +4,27 @@ import (
 	"fmt"
 
 	"github.com/ten-protocol/go-ten/go/common/container"
+	tenflag "github.com/ten-protocol/go-ten/go/common/flag"
+	"github.com/ten-protocol/go-ten/go/config"
 	enclavecontainer "github.com/ten-protocol/go-ten/go/enclave/container"
 )
 
 // Runs an Obscuro enclave as a standalone process.
 func main() {
-	config, err := enclavecontainer.ParseConfig()
+	// fetch and parse flags
+	flags := config.EnclaveFlags()
+	err := tenflag.CreateCLIFlags(flags)
 	if err != nil {
 		panic(fmt.Errorf("could not parse config. Cause: %w", err))
```

Comment:
You're reading it correctly, creation and parsing of flags are 2 separate actions. The main reason for this is how the go flags package works. Flags must be defined before they can be assigned value.

In the tests:

this errors with `no such flag -logPath`
```go
// Set the flags as needed for the test.
err := flag.CommandLine.Set(LogPathFlag, "string-value")
require.NoError(t, err)

flags := EnclaveFlags()
err = tenflag.CreateCLIFlags(flags)
require.NoError(t, err)

flag.Parse()
```

This passes because the flag is defined.

```go
flags := EnclaveFlags()
err = tenflag.CreateCLIFlags(flags)
require.NoError(t, err)

// Set the flags as needed for the test.
err := flag.CommandLine.Set(LogPathFlag, "string-value")
require.NoError(t, err)

flag.Parse()
```

wdyt ?